api/app/src/v2/Controller/DeputyController.php

```diff
@@ -2,38 +2,69 @@
 
 namespace App\v2\Controller;
 
+use App\Controller\RestController;
+use App\Entity\User;
+use App\Repository\DeputyRepository;
 use App\Repository\UserRepository;
 use App\v2\Assembler\UserAssembler;
 use App\v2\Transformer\UserTransformer;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
-use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\Annotation\Route;
 
 #[Route(path: '/deputy')]
-class DeputyController
+class DeputyController extends RestController
 {
     use ControllerTrait;
 
     public function __construct(
+        private readonly DeputyRepository $deputyRepository,
+        private readonly LoggerInterface $logger,
         private readonly UserRepository $repository,
         private readonly UserAssembler $assembler,
-        private readonly UserTransformer $transformer
+        private readonly UserTransformer $transformer,
     ) {
     }
 
-    /**
-     * @return JsonResponse
-     */
-    #[Route(path: '/{id}', requirements: ['id' => '\d+'], methods: ['GET'])]
-    public function getByIdAction($id)
+    #[Route(path: '/{id}', requirements:['id' => '\d+'], methods: ['GET'])]
+    public function getByIdAction($id): JsonResponse
     {
         if (null === ($data = $this->repository->findUserArrayById($id))) {
-            throw new NotFoundHttpException(sprintf('Deputy id %s not found', $id));
+            $this->buildNotFoundResponse(sprintf('Deputy id %s not found', $id));
         }
-
+        
+        $data = $data ?: [];
         $dto = $this->assembler->assembleFromArray($data);
         $transformedDto = $this->transformer->transform($dto);
 
         return $this->buildSuccessResponse($transformedDto);
     }
+
+    #[Route(path:'/{uid}/reports', name:'deputy_find_reports_by_uid', requirements:['uid' => '\d+'], methods:['GET'])]
+    public function getAllDeputyReports(Request $request, int $uid): JsonResponse
+    {
+        /** @var User $user */
+        $user = $this->getUser();
+        
+        if ($uid !== $user->getDeputyUid()) {
```

Comment:
Do we need to check this? Putting this constraint on the endpoint would mean that it's not usable by a superuser, for example.